pinot-core/src/main/java/org/apache/pinot/core/operator/dociditerators/MVScanDocIdIterator.java

```diff
@@ -85,9 +90,12 @@ public MutableRoaringBitmap applyAnd(ImmutableRoaringBitmap docIds) {
       int limit = docIdIterator.nextBatch(buffer);
       for (int i = 0; i < limit; i++) {
         int nextDocId = buffer[i];
-        int length = _reader.getDictIdMV(nextDocId, _dictIdBuffer, _readerContext);
-        _numEntriesScanned += length;
-        if (_predicateEvaluator.applyMV(_dictIdBuffer, length)) {
+        int[] length = new int[]{0};
```

Comment:
This seems like it can be avoided. Unnecessary object construction for every iteration in the loop ? Can we avoid passing this to doesValueMatch for pass by reference ? If this is needed to increment , then we can increment the counter in the matcher itself ?